ports/zephyr/zephyr_device.c

```diff
@@ -27,23 +27,198 @@
 #include "zephyr_device.h"
 #include "py/runtime.h"
 
+#if defined(CONFIG_MICROPY_DYNAMIC_PINCTRL)
+
+#include <zephyr/drivers/pinctrl.h>
+extern pinctrl_soc_pin_t zephyr_pinctrl_get_pinctrl(const mp_obj_t obj);
+
+#endif
+
+/* Circumvent limitation of only having devices that are initialized */
+typedef struct mp_zephyr_device_data_t_ {
+    const struct device *dev;
+    char *name;
+    const char **labels;
+    size_t label_cnt;
```

Comment:
Maybe it is possible to use the same strategy as the shell:
- https://github.com/zephyrproject-rtos/zephyr/pull/71912

Reply:
Ok thanks. Maybe something to fix upstream rather than here.